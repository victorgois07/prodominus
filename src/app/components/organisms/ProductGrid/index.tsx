import { Product } from "@/domain/entities/product.entity";
import { ProductCard } from "../ProductCard";

interface ProductGridProps {
  products: Product[];
  onEdit?: (product: Product) => void;
  onDelete?: (product: Product) => void;
}

export function ProductGrid({ products, onEdit, onDelete }: ProductGridProps) {
  if (products.length === 0) {
    return (
      <div className="text-center text-gray-500 py-8">No products found</div>
    );
  }

  return (
    <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-6">
      {products.map((product) => (
        <ProductCard
          key={product.id}
          product={product}
          onEdit={onEdit}
          onDelete={onDelete}
        />
      ))}
    </div>
  );
}
