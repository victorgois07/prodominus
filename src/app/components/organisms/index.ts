export * from "./FavoritesManagement";
export * from "./ProductCard";
export * from "./ProductFilters";
export * from "./ProductForm";
export * from "./ProductGrid";
export * from "./ProductManagement";
