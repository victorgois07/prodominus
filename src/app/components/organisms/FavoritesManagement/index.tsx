import { useFavorites } from "@/business/store/useFavorites";
import { useState } from "react";
import { Modal } from "../../molecules";
import { ProductGrid } from "../ProductGrid";

export function FavoritesManagement() {
  const { favorites } = useFavorites();
  const [isFavoritesOpen, setFavoritesOpen] = useState(false);

  return (
    <>
      <button
        onClick={() => setFavoritesOpen(true)}
        className="bg-pink-600 text-white px-4 py-2 rounded font-semibold hover:bg-pink-700 transition"
      >
        Favoritos ({favorites.length})
      </button>

      <Modal open={isFavoritesOpen} onClose={() => setFavoritesOpen(false)}>
        <h2 className="text-xl font-bold mb-4">Favoritos</h2>
        {favorites.length === 0 ? (
          <p className="text-gray-500">Nenhum produto favoritado.</p>
        ) : (
          <ProductGrid products={favorites} />
        )}
      </Modal>
    </>
  );
}
