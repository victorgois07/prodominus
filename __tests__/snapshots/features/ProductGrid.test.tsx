import { ProductGrid } from "@/app/components/organisms/ProductGrid";
import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

describe("ProductGrid Snapshot", () => {
  const mockProducts = [
    {
      id: 1,
      title: "Product 1",
      price: 99.99,
      description: "Description 1",
      category: "category1",
      image: "image1.jpg",
    },
    {
      id: 2,
      title: "Product 2",
      price: 149.99,
      description: "Description 2",
      category: "category2",
      image: "image2.jpg",
    },
  ];

  const mockOnEdit = vi.fn();
  const mockOnDelete = vi.fn();

  it("should match snapshot with products", () => {
    const { container } = render(
      <ProductGrid
        products={mockProducts}
        onEdit={mockOnEdit}
        onDelete={mockOnDelete}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with empty products", () => {
    const { container } = render(
      <ProductGrid products={[]} onEdit={mockOnEdit} onDelete={mockOnDelete} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with loading state", () => {
    const { container } = render(
      <ProductGrid
        products={mockProducts}
        onEdit={mockOnEdit}
        onDelete={mockOnDelete}
        isLoading={true}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with error state", () => {
    const { container } = render(
      <ProductGrid
        products={mockProducts}
        onEdit={mockOnEdit}
        onDelete={mockOnDelete}
        error="Failed to load products"
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with many products", () => {
    const manyProducts = Array.from({ length: 10 }, (_, index) => ({
      id: index + 1,
      title: `Product ${index + 1}`,
      price: (index + 1) * 10,
      description: `Description ${index + 1}`,
      category: `category${index + 1}`,
      image: `image${index + 1}.jpg`,
    }));

    const { container } = render(
      <ProductGrid
        products={manyProducts}
        onEdit={mockOnEdit}
        onDelete={mockOnDelete}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
