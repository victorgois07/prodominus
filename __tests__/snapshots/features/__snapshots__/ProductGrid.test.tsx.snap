// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductGrid Snapshot > should match snapshot with empty products 1`] = `
<div>
  <div
    class="text-center text-gray-500 py-8"
  >
    No products found
  </div>
</div>
`;

exports[`ProductGrid Snapshot > should match snapshot with error state 1`] = `
<div>
  <div
    class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-6"
  >
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 1"
          class="object-contain p-4"
          src="image1.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 1
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 1
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            99,99
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category1
          </span>
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 2"
          class="object-contain p-4"
          src="image2.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 2
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 2
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            149,99
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category2
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProductGrid Snapshot > should match snapshot with loading state 1`] = `
<div>
  <div
    class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-6"
  >
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 1"
          class="object-contain p-4"
          src="image1.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 1
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 1
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            99,99
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category1
          </span>
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 2"
          class="object-contain p-4"
          src="image2.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 2
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 2
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            149,99
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category2
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProductGrid Snapshot > should match snapshot with many products 1`] = `
<div>
  <div
    class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-6"
  >
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 1"
          class="object-contain p-4"
          src="image1.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 1
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 1
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            10,00
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category1
          </span>
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 2"
          class="object-contain p-4"
          src="image2.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 2
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 2
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            20,00
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category2
          </span>
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 3"
          class="object-contain p-4"
          src="image3.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 3
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 3
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            30,00
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category3
          </span>
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 4"
          class="object-contain p-4"
          src="image4.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 4
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 4
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            40,00
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category4
          </span>
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 5"
          class="object-contain p-4"
          src="image5.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 5
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 5
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            50,00
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category5
          </span>
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 6"
          class="object-contain p-4"
          src="image6.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 6
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 6
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            60,00
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category6
          </span>
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 7"
          class="object-contain p-4"
          src="image7.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 7
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 7
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            70,00
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category7
          </span>
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 8"
          class="object-contain p-4"
          src="image8.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 8
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 8
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            80,00
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category8
          </span>
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 9"
          class="object-contain p-4"
          src="image9.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 9
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 9
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            90,00
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category9
          </span>
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 10"
          class="object-contain p-4"
          src="image10.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 10
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 10
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            100,00
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category10
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProductGrid Snapshot > should match snapshot with products 1`] = `
<div>
  <div
    class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-6"
  >
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 1"
          class="object-contain p-4"
          src="image1.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 1
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 1
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            99,99
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category1
          </span>
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg shadow-md overflow-hidden hover:shadow-lg transition-shadow duration-300 relative cursor-pointer"
      role="article"
    >
      <div
        class="absolute top-2 left-2 z-10"
      >
        <button
          aria-label="Adicionar aos favoritos"
          class="font-semibold transition disabled:opacity-50 p-1 rounded-full bg-gray-100 text-gray-400"
          type="button"
        >
          ü§ç
        </button>
      </div>
      <div
        class="absolute top-2 right-2 z-10 flex gap-2"
      >
        <button
          aria-label="edit"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-gray-600"
          title="Editar"
          type="button"
        >
          ‚úé
        </button>
        <button
          aria-label="delete"
          class="font-semibold transition disabled:opacity-50 bg-gray-100 hover:bg-gray-200 rounded p-1 text-red-600"
          title="Excluir"
          type="button"
        >
          üóëÔ∏è
        </button>
      </div>
      <div
        class="relative h-48 w-full"
      >
        <img
          alt="Product 2"
          class="object-contain p-4"
          src="image2.jpg"
        />
      </div>
      <div
        class="p-4"
      >
        <h3
          class="text-lg font-semibold text-gray-800 line-clamp-2 mb-2"
        >
          Product 2
        </h3>
        <p
          class="text-gray-600 text-sm line-clamp-2 mb-4"
        >
          Description 2
        </p>
        <div
          class="flex justify-between items-center"
        >
          <span
            class="text-xl font-bold text-green-600"
          >
            R$
            149,99
          </span>
          <span
            class="text-sm text-gray-500 capitalize"
          >
            category2
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
